File: recipes_3e/apache/mysql_status/mysql_status.pl
#!/usr/bin/perl
# mysql_status.pl: print some MySQL server status information
# by generating HTML directly

use strict;
use warnings;
use Cookbook;

# Print header, blank line, and initial part of page
print <<EOF;
Content-Type: text/html

<html>
<head><title>MySQL Server Status</title></head>
<body>
EOF

# Connect to database
my $dbh = Cookbook::connect ();

# Retrieve status information
my ($now, $version) = $dbh->selectrow_array ("SELECT NOW(), VERSION()");
# SHOW STATUS variable values are in second result column
my $queries = ($dbh->selectrow_array ("SHOW GLOBAL STATUS
                                       LIKE 'Questions'"))[1];
my $uptime = ($dbh->selectrow_array ("SHOW GLOBAL STATUS
                                      LIKE 'Uptime'"))[1];
my $q_per_sec = sprintf ("%.2f", $queries/$uptime);

# Disconnect from database
$dbh->disconnect ();

# Display status report
print "<p>Current time: $now</p>";
print "<p>Server version: $version</p>";
print "<p>Server uptime (seconds): $uptime</p>";
print "<p>Queries processed:  $queries ($q_per_sec queries/second)</p>";

# Print page trailer
print <<EOF;
</body>
</html>
EOF

File: recipes_3e/apache/mysql_status/mysql_status.py
#!/usr/bin/python
# mysql_status.py: print some MySQL server status information

import cookbook

# Print header, blank line, and initial part of page

print('''Content-Type: text/html

<html>
<head><title>MySQL Server Status</title></head>
<body>
''')

# Connect to database
conn = cookbook.connect()

# Retrieve status information
cursor = conn.cursor()
cursor.execute("SELECT NOW(), VERSION()")
(now, version) = cursor.fetchone()
# SHOW STATUS variable values are in second result column
cursor.execute("SHOW GLOBAL STATUS LIKE 'Questions'")
queries = cursor.fetchone()[1]
cursor.execute("SHOW GLOBAL STATUS LIKE 'Uptime'")
uptime = cursor.fetchone()[1]
q_per_sec = "%0.2f" % (float(queries) / float(uptime))
cursor.close()

# Disconnect from database
conn.close()

# Display status report
print("<p>Current time: %s</p>" % now)
print("<p>Server version: %s</p>" % version)
print("<p>Server uptime (seconds): %s</p>" % uptime)
print("<p>Queries processed: %s (%s queries/second)</p>"
      % (queries, q_per_sec))

# Print page trailer
print('''
</body>
</html>
''')
